js/edit_delete.js

```diff
@@ -0,0 +1,116 @@
+import { request } from "../js/HTTP_request_base.js"; // Import the request function
+import { createPost, clearModalContent } from "./create.js";
+
+const editPostModal = new bootstrap.Modal(
+  document.getElementById("createPostModal")
+);
+
+let currentEditingPostId = null;
+
+////////// EDIT POST
+async function editPost(data, postId) {
+  const url = `https://api.noroff.dev/api/v1/social/posts/${postId}`;
+  try {
+    const result = await request(
+      url,
+      "PUT",
+      data,
+      localStorage.getItem("token")
+    );
+    if (!result) {
+      throw new Error("Error editing post");
+    }
+    return result;
+  } catch (error) {
+    console.error("Failed to edit post:", error);
+    throw error; // Re-throw the error to propagate it to the caller
+  }
+}
+
+////////// DELETE POST
+async function deletePost(postId) {
+  const url = `https://api.noroff.dev/api/v1/social/posts/${postId}`;
+  try {
+    const result = await request(
+      url,
+      "DELETE",
+      null,
+      localStorage.getItem("token")
+    );
+    if (!result) {
+      throw new Error("Error deleting post");
```

Comment:
Same thing here

Reply:
Should now be showing inline error messages for the user to see.